dev/snippets/lib/snippets.dart

```diff
@@ -93,6 +93,62 @@ class SnippetGenerator {
         return injections[componentIndex].mergedContent;
       }
     }).trim();
+    return _sortImports(interpolated);
+  }
+
+  String _sortImports(String code) {
+    final List<String> result = <String>[];
+    final List<String> lines = code.split('\n');
+    final List<String> imports = <String>[];
+    int firstImport = -1;
+    int lineNumber =0;
+    for (final String line in lines) {
+      if (RegExp(r'^\s*import').matchAsPrefix(line) != null) {
+        if (firstImport < 0) {
+          firstImport = lineNumber;
+        }
+        imports.add(line);
+      } else {
+        result.add(line);
+      }
+      lineNumber++;
+    }
+    if (firstImport > 0) {
+      final List<String> dartImports = <String>[];
+      final List<String> packageImports = <String>[];
+      final List<String> otherImports = <String>[];
+      final RegExp typeRegExp = RegExp(r'''import\s+['"](?<type>\w+)''');
+      imports.sort();
+      for (final String importLine in imports) {
```

Comment:
It's unfortunate to see this logic duplicated here but I don't really see any other way. 🤔

Reply:
FWIW, here's where the logic lives in analysis server: 

https://github.com/dart-lang/sdk/blob/cc1f9b9564144ca5d40bbe21b88acfcd461cf56a/pkg/analysis_server/lib/src/services/correction/organize_imports.dart#L65

Reply:
Oh, thank you! The implementation here is pretty dumb, and won't survive things like multi-line "show" or "hide" statements, and will mangle comments attached to the import line.

I did it this way here because it's just the samples, and they're pretty normalized, but Ian and I decided a while ago that we didn't want to add the analyzer as a dependency, so I have another implementation of the snippets library that lives in the `assets-for-api-docs` repo. I am going to implement this there using the analyzer, and seeing that organize imports code really helps with that. Eventually, the snippets tool will live there.

Can I use the `ImportOrganizer` directly, or is that only part of the server implementation?

Reply:
Looks like it's just part of the server, and the `analysis_server` package seems to not be null-safe yet (requires sdk < 2.0.0), but I think I can adapt that code for the "good" implementation.

Reply:
fyi @scheglov @bwilkerson

Reply:
For posterity, here's the "real" implementation that will be used once we switch over to the separate snippets package: https://github.com/flutter/assets-for-api-docs/pull/145